load-test-framework/src/main/java/com/google/pubsub/flic/output/SheetsService.java

```diff
@@ -0,0 +1,125 @@
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.SheetsScopes;
+import com.google.api.services.sheets.v4.model.ValueRange;
+import com.google.pubsub.flic.common.LatencyDistribution;
+import com.google.pubsub.flic.controllers.Controller;
+import com.google.pubsub.flic.controllers.Client;
+import com.google.pubsub.flic.controllers.Client.ClientType;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.LongStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SheetsService {
+  private final static Logger log = LoggerFactory.getLogger(SheetsService.class);
+  private Sheets service;
+  
+  private final String APPLICATION_NAME = "loadtest-framework";
+  
+  public SheetsService(String dataStoreDirectory) throws Exception {
+    InputStream in = new FileInputStream(new File(System.getenv("GOOGLE_OATH2_CREDENTIALS")));
+    JsonFactory factory = new JacksonFactory();
+    GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(factory, new InputStreamReader(in));
+    HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport(); 
+    FileDataStoreFactory dataStoreFactory = new FileDataStoreFactory(new File(dataStoreDirectory));
+    List<String> scopes = Arrays.asList(SheetsScopes.SPREADSHEETS);
+    GoogleAuthorizationCodeFlow flow =
+        new GoogleAuthorizationCodeFlow.Builder(
+                transport, factory, clientSecrets, scopes)
+        .setAccessType("offline")
+        .setDataStoreFactory(dataStoreFactory)
+        .build();
+    Credential credential = new AuthorizationCodeInstalledApp(
+        flow, new LocalServerReceiver()).authorize("user");
+    service = new Sheets.Builder(transport, factory, credential)
+        .setApplicationName(APPLICATION_NAME).build();
+  }
+  
+  /* Publishes stats information to Google Sheets document. Format for sheet assumes the following
+   * column order: Publisher #; Subscriber #; Message size (B); Test length (s); # messages;
+   * Publish batch size; Subscribe pull size; Request rate; Max outstanding requests; 
+   * Throughput (MB/s); 50% (ms); 90% (ms); 99% (ms)
+   */
+  public void sendToSheets(String sheetId, Map<ClientType, Controller.LoadtestStats> results) {
+    List<List<Object>> cpsValues = new ArrayList<List<Object>>(results.size());
+    List<List<Object>> kafkaValues = new ArrayList<List<Object>>(results.size());
+    results.forEach((type, stats) -> {
```

Comment:
I think it would be a good bit harder to adapt to different spreadsheet styles, and once you require that knowledge, I don't think the usage gains from not needing to know result types actually helps you that much. I'll keep this in mind as a todo, but I don't think it's worth the time right now.